### PR TITLE
add speech recognition using gluon

### PR DESCRIPTION
**Pull request by mengjiexu**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: example/gluon/gluon_speech_recognition/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+mxnet 语音识别 gluon
```
Comment (ThomasDelteil) [inline example/gluon/gluon_speech_recognition/README.md]: Please write the tutorial in English

File: example/gluon/gluon_speech_recognition/HybridSequential_CNN_CTC.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+def get_iter(batch_size):
+    data_dir = './data'
+    train_iter = get_data_gen(data_dir, get_str2idx(data_dir)[1], batch_size)
+    for x, y in train_iter:
+        yield nd.array(x), nd.array(y)
+
+
+class ShowProcess():
+    """
+    显示处理进度的类
```
Comment (ThomasDelteil) [inline example/gluon/gluon_speech_recognition/HybridSequential_CNN_CTC.py]: The comments and description needs to be in English

**Comment by ThomasDelteil** [comment]
@mengjiexu, thanks for your contribution. Please add a description of the PR, your example needs to point to a publicly accessible dataset, and provide instruction on how to load the data. You also need to translate all the chinese comments / description to english. For Chinese only content, please refer to https://github.com/mli/gluon-tutorials-zh

File: example/gluon/gluon_speech_recognition/HybridSequential_CNN_CTC.py (hunk 2 above)
Comment (mengjiexu) [reply example/gluon/gluon_speech_recognition/HybridSequential_CNN_CTC.py]: ok

**Comment by szha** [comment]
@Jerryzcn

**Comment by stu1130** [comment]
@mxnet-label-bot[pr-awaiting-review]

**Comment by kobenaxie** [comment]
HI @mengjiexu ,could you please release your ASR result , like CER or WER ?

**Comment by mengjiexu** [comment]
I am training the model, need one or two days to train and test the model.

**Comment by vrakesh** [comment]
@mengjiexu Requesting an update on this PR, is this ready to merge?

**Comment by mengjiexu** [comment]
I have used database thchs30 to train this model. In the early stage, the loss go down from 800 to 200,
( **I  tested the model when the loss is the lowest, the word accuracy almost 30%**) and after a few hours, it began to go up to thousand and million and at  the end nan. I have tried a lot of methods to train the model, such as using very small lr, using very deep network, using very small network, they are all failed. And so, I can't make sure that the network is good at the ASR work.

**Comment by mengjiexu** [comment]
Is there anyone can help me? Thanks a lot.

**Pull request closed without merging**